modal/_utils/shell_utils.py

```diff
@@ -17,16 +17,23 @@
 def write_to_fd(fd: int, data: bytes):
```

Comment:
We could probably simplify this to just register a writer with `add_writer` once, but that requires a bigger refactor